### ISSUE TITLE
GestureDetector doesn't respond correctly after transition with Transform widget

### ISSUE THREAD

**Issue by MariaMelnik** [flutter/flutter#27587]
Transformed widget (GestureDetector with inner child) doesn't respond correctly.
Part of widget what stays in original boundaries is good, but outside original boundaries - doesn't respond.
Flutter inspector shows correct boundaries of transformed widget.

main.dart with this issue: [https://gist.github.com/MariaMelnik/a8a3cfaeaf7dbc11ead0dd472790040d](url)

flutter inspector:
![image](https://user-images.githubusercontent.com/20188655/52337667-52032e80-2a19-11e9-88ed-5f6e472e696f.png)

flutter doctor:
```
[✓] Flutter (Channel beta, v1.0.0, on Mac OS X 10.13.6 17G65, locale en-RU)
    • Flutter version 1.0.0 at /Users/mariamelnik/work/flutter
    • Framework revision 5391447fae (10 weeks ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[!] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/mariamelnik/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[!] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    ✗ Verify that all connected devices have been paired with this computer in Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.2
    ✗ ios-deploy out of date (1.9.4 is required). To upgrade with Brew:
        brew upgrade ios-deploy
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 32.0.1
    • Dart plugin version 182.5124
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] IntelliJ IDEA Community Edition (version 2017.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 26.0.1
    • Dart plugin version 173.4700

[✓] Connected device (1 available)
    • iPhone X • 105FE9DF-AE89-4F8F-BFFC-D920913E2EAA • ios • iOS 11.4 (simulator)
```

**Comment by Kartik1607** [comment]
When you tap on part of widget outside the original boundary lines, the **parent widget thinks you are not hitting it** (in your case the 200 width container) since, you technically tapped outside the widget.  Then it no longer checks for hit test on child widgets too. Issue has been discussed https://github.com/flutter/flutter/issues/6606 before. I am curious how a transformed widget should receive hit tests though.  @Hixie could you please explain, how should one be able to get hit tests on a transformed widget. I saw one answer [here](https://stackoverflow.com/questions/51366761/in-flutter-how-can-a-positioned-widget-feel-taps-outside-of-its-parent-stack-ar) but shouldn't it be more straight forward?

**Comment by MariaMelnik** [comment]
Kartik1607, I saw #6606. There GestureDetector outside Transformed widget. But in my case it is inside.
Also I saw [https://stackoverflow.com/questions/51366761/in-flutter-how-can-a-positioned-widget-feel-taps-outside-of-its-parent-stack-ar](this question). There are widgets in Stack what I think little bit different case.

**Comment by Hixie** [comment]
`Transform` is a bit weird in that it just passes the hit tests through unmodified. So if it transforms its child within the bounds of the parent, then hit tests will work even if they are transformed outside the bounds of the `Transform` (as seen by the parent) (assuming the parent passes all hits inside its region to all its children, which is the default behaviour).

**Comment by MariaMelnik** [comment]
[Hixie](/Hixie), I don't have Transform in GestureDetector (if I understand you are talking about this situation). My case opposite: I have GestureDetector in Transform. So I think Transform shouldn't pass anything because it is on upper layer. Am I wrong?
I attach picture. Green rectangle gets taps, but red does not. Both of this rectangles are inside GestureDetector (as you can see on my start screenshot).
<img width="341" alt="screen shot 2019-02-06 at 14 12 26" src="https://user-images.githubusercontent.com/20188655/52376191-81925500-2a72-11e9-8609-0213ca5842f5.png">

**Comment by Hixie** [comment]
What matters[1] is that the child of the Transform (whatever that is) is inside the bounding box of the parent of the Transform (whatever that is). It doesn't actually matter which is the gesture detector, the gesture detector just defers to its child for determining whether it's hit or not[2].

[1]: assuming you're using only widgets from the framework. It's possible to create widgets that act differently and if they're involved then what I'm saying may be wrong.

[2]: unless you change the default hit test behaviour on the gesture detector.

**Comment by Kartik1607** [comment]
So if a Transform widgets translates a widget outside the bounds of parent widget, what would a developer need to do to in order to register a hit? Could this problem be solved if the hit test chain was not stopped when clicked outside patent widget but rather the whole widget tree is hit tested, but this would just increase complexity to O(n) instead Of(logn)? One more option I can think of is to have a different hittest bounds parameter for each widget  node which can be updated by child widgets (like transform would increase hittest bound for container) which would then check hit for container using its original size and then defer to children. ?

**Comment by Hixie** [comment]
> So if a Transform widgets translates a widget outside the bounds of parent widget, what would a developer need to do to in order to register a hit?

Make the parent bigger.

> Could this problem be solved if the hit test chain was not stopped when clicked outside patent widget but rather the whole widget tree is hit tested, but this would just increase complexity to O(n) instead Of(logn)?

Other hit testing strategies would have different tradeoffs, and some could find hits to nodes that are outside their ancestors, yes.

> One more option I can think of is to have a different hittest bounds parameter for each widget node which can be updated by child widgets (like transform would increase hittest bound for container) which would then check hit for container using its original size and then defer to children. ?

Generally I would recommend refactoring the tree so that you don't transform something that you want to hit test outside of its ancestors' bounds. For example, when we had a similar problem with the floating buttons that come when you tap on text fields, we used the Overlay to put the toolbar outside the TextField.

**Comment by zoechi** [comment]
Similar #25702, #18408

**Comment by MariaMelnik** [comment]
[Hixie](/Hixie) thank you for explanation.
I can assume case with many widgets in one screen where each of them should respond on hit with some transformation (size, translate, etc.). How much will be this transformation we know only in runtime. If I use Transform, I should put every widget into container as big as screen size. It means stack with many containers. It can be hard (or maybe even impossible) to provide gestures to target widget through all stack's children.
What can you recommend to solve it? Should we avoid Transform in such cases?

**Comment by joseph-montanez** [comment]
I just hit this feature? I moved away from the transform and leverage animating the container width, height, margins/paddings to do transforms. I even tried my transforms in reverse thinking if the transform started where it was going to end up and then I reverse() the animation controller, that it would work... nope. Just seems weird, you want to animate a button moving up and it doesn't matter if you have the gesture detector inside or outside the transform, it simply does not work.

**Comment by joseph-montanez** [comment]
I just wanted to ping with a response, I recently did a flutter upgrade. For some reason, while Flutter did not upgrade, something else did and has fixed this behavior. I also switched from the stable channel to the beta channel and noticed it worked there as well. Maybe the switching of the channels made it work, but I did switch back to stable and it continued to work.

**Comment by DatChoob** [comment]
I have the same issue while replicating from this site. None of the surrounding buttons are clickable. https://fireship.io/lessons/flutter-radial-menu-staggered-animations/

**Comment by rahulthakur319** [comment]
As mentioned by @Hixie , Making the parent bigger solved the problem for me

```
return Stack(
          overflow: Overflow.visible,
          children: <Widget>[
            Positioned(
              bottom: 0,
              width: MediaQuery.of(context).size.width,
              height: MediaQuery.of(context).size.height,
              child: <MyCustomStackWithAnimations>
....
```

**Comment by kaustavjaiswal** [comment]
For anyone still stuck with this issue, making the parent bigger does solve the problem as mentioned by @Hixie

**Comment by dannyvalentesonos** [comment]
In situations where an image is to be used, for example, that is placed inside a parent that has a fixed size that you cannot control, but want to give it a slightly larger gesture detector to register hits that are larger than the image, what is one supposed to do? I've used an Overflow box, which allows the bounds of the gesture detector to grow, but that still doesn't allow hits to be detected outside the original box of the parent. Is there any solution for that kind of scenario?

**Comment by Hixie** [comment]
@dannyvalentesonos Questions are best asked on a site like Stack Overflow or in the flutter-dev mailing list. This is an issue database, which is about bugs and feature requests. If you want to file a bug or issue request, the right place for that is https://github.com/flutter/flutter/issues/new/choose.

**Comment by Riki432** [comment]
I was having this problem. I had buttons that slid out from under another button. The buttons were stored in a stack and onPress wasn't triggering. I just put a container across the whole width of the screen as the first element on the stack and it worked. I'm sure this isn't the most elegant solution though.

**Comment by tb2761** [comment]
In my oppinion this must by fixed with next release, because i exspect, that GestureDetector check if the tap was in the bounds where the childs placed.
But actually it check if the tap was in the bounds where the child at first build placed and every moving of the childs are ignored.
This is in my oppinion a big issue which make the GestureDetector in most cases completely useless.

**Comment by Vardiak** [comment]
@Hixie Hi!
I'm working on a 2D adaptation of game which is already available on the web with 100K+ registered users. After multiple months of using this framework, I really feel like this is the most serious issue.
I've already encountered it in more than 5 scenarios where I had to refactor everything to make the parent bigger, which made the code very hard to understand.
I am once again facing this issue and it will just be way too hard to overcome this time. I will have to change my UX because I can't do it in Flutter (changing the widget size would affect its alignement which needs to be very precise).
I know that there is a performance reason for this, but please, we need an option to have a global hit detection that do no rely on the widget tree. A transformed GestureDetector should work fine without needing the parent to contain the transformed element.
Best regards,

**Comment by ukomiljon** [comment]
I am also facing this probem that after parent making smaller the child gasture does not work, after makeing position width and height bigger it works, but after more scale it child's gasture does not work again.

**Comment by ukomiljon** [comment]
I have zoomable and moveable container, inside the container there are draggable containers. We cannot make draggable containers while gesture is problem

**Comment by dgtlmonk** [comment]
Flutter newbie here and I ran across this same situation; I did a quick test with Positioned class instead of Transform. Looks like it's a good alternative  if you want GestrureDetector to work

```
Column(
        children: <Widget>[
          Stack(
            children: <Widget>[
              Container(width: 120, height: 120, color: Colors.lightBlue),
              Positioned(
                top: 80.0,
                width: 100,
                height: 50,
                child: GestureDetector(
                    onTap: () {
                      print('ok');
                    },
                    child: Container(
                      color: Colors.purple,
                      child: Text(
                        'Clickety Click',
                        style: TextStyle(color: Colors.white),
                      ),
                    )),
              )
            ],
          ),
        ],
      ),

```

**Comment by esDotDev** [comment]
@Hixie Would you guys be open to a pull request on this that adds the option for developers?

Seems like it could be rather simple.
* Add `Stack.allowHitTestOutsideBounds` and `RenderBox.allowHitTestOutsideBounds`
* Modify Stack to set `allowHitTestOutsideBounds `on it's RenderStack
* Modify RenderBox to respect the `allowHitTestOutsideBounds` flag

I would be really nice to have the flexibility to be able to offset things outside their bounds, and have hit detection. I understand the performance worries, but in a lot of cases, we have performance to spare, and just need the functionality.

**Comment by Vardiak** [comment]
> @Hixie Would you guys be open to a pull request on this that adds the option for developers?
> 
> Seems like it could be rather simple.
> 
> * Add `Stack.allowHitTestOutsideBounds` and `RenderBox.allowHitTestOutsideBounds`
> * Modify `Stack` to set `RenderStack.allowHitTestOutsideBounds`
> * Modify `RenderBox` to respect the `allowHitTestOutsideBounds` flag
> 
> I would be really nice to have the flexibility to be able to offset things outside their bounds, and have hit detection. I understand the performance worries, but in a lot of cases, we have performance to spare, and just need the functionality.

I'm not an expert, but if I understood everything correctly I don't think this would be the right approach for multiple reasons :

- This problem isn't always caused by a Stack but often a simple Transform widget.
- You would need to pass all pointer events to every widget of the widget tree, which in this case is very costly.

In my opinion, the best user experience would be that we add a `GlobalHitDetection` widget, which would calculate the bounds of its child, and absorb every pointer event in it, and redirect it to its child. This would allow us to make global hit detection work for `ListView` and every widget of this framework that uses pointer events. The only drawback with this technique is that we would need to calculate the bounds at every frame, since the widget may have been transformed 😞

**Comment by esDotDev** [comment]
Yes, good point, I had a specific use-case in mind, but this would be a significantly deeper fix to handle all use cases.

**Comment by esDotDev** [comment]
Here's a simple reproduction case for this issue:
```
class WidgetClipIssueSpike extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Stack(
          children: <Widget>[
            Transform.translate(
                offset: Offset(20, 100),
                child: GestureDetector(
                  onTap: (){ print("Why is this not firing? Change offset to 0,0 and I will work(?)"); },
                  child: Container(
                    width: 100,
                    height: 100,
                    color: Colors.red,
                  ),
                ))
          ],
        ),
      ),
    );
  }
}
```

The red square should be clickable but is not. Set offset to 0,0 and the gesture works. Also, wrapping the GestureDetector Child, in a top-left Align(), also will work but is quite counter-intuitive (you wouldn't think to Align a Positioned element)

**Comment by gskinner** [comment]
This also impacts scrolling interactions - both touch scrolling and mousewheel. In the following example, a multiline `Text` field is constrained in a 100px tall `SizedBox` and made scrollable with `SingleChildScrollView`. The `SizedBox` is then translated 60px down.

Only the top 40px of the `Text` will register scroll interactions (ie. the intersection between the transformed and untransformed bounding boxes).

```dart
import 'package:flutter/material.dart';
import 'dart:ui' as ui;

class Test1 extends StatelessWidget {
  Test1({
    Key key,
  }) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xffffffff),
      body: Stack(
        children: <Widget>[
          Transform.translate(
            offset: Offset(0.0, 60.0),
            child: SizedBox(
              width: 357.0,
              height: 100.0,
              child: SingleChildScrollView(
                  child: Text(
                'This field is a test of scrolling text.\nNote that scrolling doesn’t work in transformed widgets.\nLine 3\nLine 4\nLine 5\nLine 6\nLine 7\nLine 8\nLine 9\nLine 10 (last line)',
                style: TextStyle(
                  fontFamily: 'Helvetica Neue',
                  fontSize: 20,
                  color: const Color(0xffff0000),
                ),
                textAlign: TextAlign.left,
              )),
            ),
          ),
        ],
      ),
    );
  }
}
```

**Comment by iapicca** [comment]
consolidating from
https://github.com/flutter/flutter/issues/18408
https://github.com/flutter/flutter/issues/49210
https://github.com/flutter/flutter/issues/52171

**Comment by Vardiak** [comment]
I guess that the Flutter team isn't willing to fix this issue for now, since they consider that this an edge, non recommended case. I can assure you that this is more common than you might think and that there isn't always a straight forward solution.

**Comment by esDotDev** [comment]
Indeed. It's not that there isn't always some alternate architecture you could use, it's just almost always not the way you naturally want to do it, forcing you into more elaborate, de-coupled designs that really just serve to create complexity in the codebase, which in turns leads to more bugs and harder debugging.

**Comment by Vardiak** [comment]
> Indeed. It's not that there isn't always some alternate architecture you could use, it's just almost always not the way you naturally want to do it, forcing you into more elaborate, de-coupled designs that really just serve to create complexity in the codebase, which in turns leads to more bugs and harder debugging.  

I sometimes end up using a LayoutBuilder and a Stack, and handle all my layout by hand. The code gets unreadable, but it works 😬

**Comment by timsneath** [comment]
Added product tag, since this is related to a demo @esDotDev and team are building for us, c.f. https://github.com/flutter/flutter/issues/27587#issuecomment-612461617 in particular.

**Comment by Hixie** [comment]
My comments above starting at https://github.com/flutter/flutter/issues/27587#issuecomment-461242787 describe what is going on here and how to design one's application to avoid running into this. This is basically working as intended. I don't think there's anything to change here.

**Comment by Hixie** [comment]
(closing per last comment; please don't hesitate to reach out on our chat server if you would like to discuss this)

**Comment by gskinner** [comment]
@Hixie - Thanks for your comments.

I think the concern in this thread is not whether it is "working as intended" from an engineering perspective, but whether it is "working as _expected_" from an implementation perspective. The comments here suggest latter is not true. 

As many people have pointed out, this unexpected limitation introduces significant inflexibility into a framework that is awesome _because_ of its flexibility. It forces developers to abandon innovative UI concepts, and/or implement complicated, fragile workarounds that are less semantically clear and more expensive to maintain.

"Working as intended" doesn't help the community to understand why it doesn't work "the way it should". Is it too difficult to implement? Is it too costly from a performance perspective in specific scenarios? If the latter, what are those scenarios? Is there a way that Flutter could let us signal our intent so the default is fast, but I can opt in to a slower approach that solves these issues?

With more information, we might be able to help with ideas or pull requests. This issue will almost certainly keep coming up, as it's not expected that there will be a class of layout widget that breaks interactions.

**Comment by Hixie** [comment]
Sorry for the lack of clarity, I can see how that would be frustrating. Let me see if I can elucidate the reasoning a bit.

The design of hit-testing is intended to be O(N) in the depth of the tree, essentially minimizing the amount of the tree that has to be visited and the maths that must be performed at each step. The first assumption we use to do this is that if you don't hit a node, then you can't hit its children. That means that, for example, when we are inspecting a box at position 100,100 with size 50x50, we know that if the hit is for pixel 20,20 there is no way the box could be involved and we short-circuit the walk at that point, skipping the entire subtree. (The other assumption is that hit testing is based on points, not regions. That isn't as relevant here, but affects some other limitations we see in hit testing relative to some of the use cases and feature requests we see.)

To make it possible to hit nodes that overflow their ancestors, e.g. because Transform moves them, we would have to walk every node in the entire tree when doing a hit test. This would be significantly more expensive.

In general the workarounds here are relatively simple. Using the Overlay is the most common solution, and is what is used e.g. for the text editing toolbar. It essentially side-steps the issue by putting the transformed content into an ancestor that is big enough.

**Comment by Vardiak** [comment]
@Hixie Hi!
Thank you for the clarification, it was much needed. I understand the performance implication of passing down all pointer events to all widgets. But would it be possible to implement a global hit detection in parallel of the tree-based hit detection algorithm currently used? We would need to wrap a widget so that it absorb all pointer event within its real bounds and then pass it down to sub widgets using the usual algorithm. It seems like a simple way to bypass this problem without needing to refactor too much code when encountered. Is this technically possible?

By the way, why is hit detection so much more precise on the web? How does it work? Does it come at a performance cost?

**Comment by Hixie** [comment]
You could create an object that does hit testing as you suggest by having it register with the PointerRouter, yes. I wouldn't recommend it (it would be inconsistent with respect to overlaps with other similar widgets); I would instead recommend using the Overlay instead.

The Web has much worse performance.

**Comment by esDotDev** [comment]
Certainly once you know the ins and outs of Overlay it can work as a workaround in many cases, however there are a couple points on that:
1. Most Flutter developers I've seen do not fully understand Overlay, how to use it, and how to position objects to stick to a parent. There's a substantial learning curve to simple making a button overhang it's parent by a few pixels. I don't think that's ideal.

2. Even then, it will cause side effects and unwanted issues. For example we're building our own dropdowns now, and FocusTraversalGroups don't work, since the Widgets is not actually in the Tree, and it has rendering issues when the target sorted behind other items (since it's in an overlay)
https://github.com/flutter/flutter/issues/55016#issuecomment-615147832

And those are just the two we've discovered in a week or two of working with it. I would guess there's quite a few more in other edge case scenarios as we continue to push it in different directions since it's such a extreme fix to the problem.

**Comment by Hixie** [comment]
If there's a specific use case that doesn't work, my recommendation would be to file a bug just about that specific case, rather than asking for a specific solution. There's probably better ways to solve the problems that are leading to this request.

**Comment by esDotDev** [comment]
Ok. We have some specific issues related to Overlay that we would love to get fixed, and will try an log a bug soon. This issue in particular is not blocking the demo project mentioned above. 

This discussion basically boils down to feedback that Overlay is (relatively) hard to use, and being able to Transform manually, both within, and without their parent, would be a great gift to flexibility and speed of development for all Flutter devs. 

It just enabled those simple little use cases, where Overlay is overkill, and avoids having to rethink your entire responsive app scaffold because a widget far down in the tree needs some particular offset by 10px or something.

**Comment by gskinner** [comment]
@Hixie - I really appreciate the additional information, it helps a lot with understanding this issue.

I don't mean to second guess you, but do you know if there has been any specific performance testing on this? I would expect that running a bounds test against a point for every GD in the tree would be very cheap. The run time would be O(N), you just wouldn't have as many opportunities to exit early (to reduce N). Perhaps if you get into specific high-frequency, high-count interactions like multitouch dragging cost becomes an issue?

This is in no way a 1 to 1 comparison, but in EaselJS, we were able to run _much_ more expensive hit testing (pixel tests without bounds checks) across the whole tree without slowdowns. Similarly, browsers are able to make this work with a much more complex and messy display tree. For clarity, I'm not suggesting that means this should be true for Flutter, just making a comparison to things I know well.

Given that this likely has been tested and found to cause problems in real world applications, there might still be a couple of ways to address this, though I haven't looked into the implementation challenges, and I admit they may be naive:

1. Include transformed widgets in the parent's hit bounds. I'm guessing that these bounds are shared with layout or other processes though. If so, this may be a non-starter.
2. Provide some way to hint to the framework that you'd like to bypass culling on the parent. I'm not sure as to specifically how this could be implemented (or even if it can), but you could imagine something like (names TBD) `Transform(enableGestures: true, ...)`, or even a widget that causes its immediate child (or all descendants?) to ignore the bounds check `IgnoreBoundsCheck(child: MyParent(child: Transform(...))`.

Thanks again for your attention to this. Having implemented (much simpler) versions of this in the past, I have a deep appreciation for the complexity involved, and the thought that goes into trying to make it work both fast and well.

**Comment by Hixie** [comment]
> I would expect that running a bounds test against a point for every GD in the tree would be very cheap.

Well, the Gesture Detectors themselves don't do any hit testing. It would be the RenderPointerListeners. But those don't know where they are in the tree. You'd have to walk every ancestor to figure out where those nodes are. You'd also need to make sure you clip to RenderClipRects and other clips. You'd also need to project through any 3D transforms (e.g. in RenderTransforms) to make sure the pointer is correctly hitting the right node. Also you need to do all this in a way that still works when only a small subset of the tree gets dirtied between frames. And now I've just described the entire algorithm that we do. :-)

It's possible that we could spend more CPU cycles doing something different. But then we'd need to remove the CPU cycles from something else so as to not be a net negative on the whole system.

I encourage you to file specific issues on the specific use cases you have. It's not really useful to discuss this in the abstract because there's no way to actually evaluate the proposals.

**Comment by esDotDev** [comment]
In case it got lost in the shuffle, there is this specific issue:
https://github.com/flutter/flutter/issues/54551

Seems very counter-intuitive that parent _size_ is enforced, when transforming outside the parent _bounds_ is not, and then on top of that, scale() somehow sidesteps the bound-check.

**Comment by louis97** [comment]
Similiar issue found here, but with TextField. https://stackoverflow.com/questions/63140805/textfield-wrapped-in-a-transform-translate-does-not-focus-when-clicked/63141239?noredirect=1#comment111658258_63141239

**Comment by br-programmer** [comment]
I have a similar problem, The Gesture Detector does not respond when I apply a Transform.translate () Apparently it only moves the child. I am using a Grid. It is impossible for me to enlarge the size of the parent of each item

![chairs](https://user-images.githubusercontent.com/30538983/95106275-ebf31100-06fd-11eb-9c4f-35bdbad19856.png)

**Comment by tranice** [comment]
I also meet the same problem, when I rotate a marker on the map , it can't be tapped. I spent a lot of time to track bug, finally I came here.  needs is served by tech, isn't it?

**Comment by kartik1225** [comment]
Any updates on this?

This is also not working with the negative `Positioned` widget.

**Comment by Vardiak** [comment]
> @Hixie Hi!
> I'm working on a 2D adaptation of game which is already available on the web with 100K+ registered users. After multiple months of using this framework, I really feel like this is the most serious issue.
> I've already encountered it in more than 5 scenarios where I had to refactor everything to make the parent bigger, which made the code very hard to understand.
> I am once again facing this issue and it will just be way too hard to overcome this time. I will have to change my UX because I can't do it in Flutter (changing the widget size would affect its alignement which needs to be very precise).
> I know that there is a performance reason for this, but please, we need an option to have a global hit detection that do no rely on the widget tree. A transformed GestureDetector should work fine without needing the parent to contain the transformed element.
> Best regards,

I posted this comment over a year ago. Since then, anything has changed (beside my app scaling to 2M users), it is still a pain to work on custom UI with complex user gestures... I would really like the creation of a global hit detection algorithm to be reconsidered, it would make our code much more maintainable. Yes, this issue can be countered in most cases but it often ends up with very ugly code where there would have been a very elegant solution with global hit detection.

This, with the janks on iOS are for me the 2 most important issues in Flutter that makes it really hard to work with.

**Comment by Hixie** [comment]
This bug is closed, so it's unlikely anything will happen here. I encourage you to file specific issues on the specific use cases you have.

**Comment by esDotDev** [comment]
> > I've already encountered it in more than 5 scenarios where I had to refactor everything to make the parent bigger, which made the code very hard to understand.

This issue seems to be a nice elegant way to get this done:
https://github.com/flutter/flutter/issues/75747 

How do we further elevate this? I know the answer is typically upvotes, but pretty sure there have been many issues, with many upvotes, on this general topic that are now closed.
